electionleaflets/apps/core/s3_thumbnail_store.py

```diff
@@ -41,11 +41,12 @@ def _get_thumbnail_filename(self, source, geometry_string, options):
             ["{}={}".format(k, v) for k, v in list(opts.items())]
         )
 
-        return (
-            "{base_url}/{geometry_string}/{url_kwargs}/{original_path}".format(
-                base_url=base_url,
-                geometry_string=geometry_string,
-                url_kwargs=url_kwargs,
-                original_path=source.name,
-            )
+        file_path_parts = source.name.rsplit(".", 1)
+        file_name = file_path_parts[0]
```

Comment:
There's a number of places peppered around this code where we are using `.rsplit(".", 1)` to separate into filename and extension (accounting for multiple dots in the filename). Two things here:

1. I think it would be useful to include a testcase with a filename that has multiple dots in it (e.g: `local.buckinghamshire.abbey.2025-05-01.tiff`), just to make sure we aren't introducing bugs around this in future
2. This is more of a readability thing than a correctness thing, but `pathlib` has quite a nice abstraction for this:

```py
>>> from pathlib import Path
>>> p = Path('local.buckinghamshire.abbey.2025-05-01.tiff')
>>> p.stem
'local.buckinghamshire.abbey.2025-05-01'
>>> p.suffix
'.tiff'
```